docs/design/processing_trace_data/records_service.md

```diff
@@ -0,0 +1,214 @@
+
+For the intermediate columns, the time at the intermediate point between start and end is represented.
+
+This table representation can be used for various measurement targets such as callbacks, nodes, and paths.
+
+The rows of the table are visualized as lines in a message flow diagram.
+
+This section describes the process o
+The class that reads TraceData is the Records object.
+
+- [Records](./records.md)
+
+The following are processing using Records object.
+
+- [Period](./records_service.md#period)
+- [Frequency](./records_service.md#frequency)
+- [Latency](./records_service.md#latency)
+- [Response time](./records_service.md#response-time)
+
+See also
+
+- [Latency definition](../latency_definitions/index.md)
+
+See also
+
+- [Records](./records.md)
+
+## Period
+
+The period is defined as the time difference between the columns of interest.
+
+$$
+    period_n = t_{n} -t_{n-1}
+$$
+
+### Example
+
+Input
+
+| Start timestamp |
+| --------------- |
+| 0.0             |
+| 1.0             |
+| 2.0             |
+| 3.0             |
+| ...             |
+
+Output
+
+| Timestamp | Period |
+| --------- | ------ |
+| 0.0       | 1.0    |
+| 1.0       | 1.0    |
+| 2.0       | 1.0    |
+| ...       | ...    |
+
+See also
+
+- [API Reference | Period](https://tier4.github.io/CARET_analyze/latest/record/#caret_analyze.record.Period)
+
+## Frequency
+
+Frequency is defined as the number of events that occur in one second.
+
+### Example
+
+Input
+
+| Start timestamp |
+| --------------- |
+| 0.0             |
+| 0.1             |
+| 0.5             |
+| 1.2             |
+| 1.3             |
+| 2.3             |
+| ...             |
+
+Output
+
+| Timestamp | Period |
+| --------- | ------ |
+| 0.0       | 3.0    |
+| 1.0       | 2.0    |
+| ...       | ...    |
+
+## Latency
+
+Latency is defined as the time difference between the two columns of interest.
+
+$$
+latency_n = t^{end}_{n} - t^{start}_{n}
+$$
+
+### Example
+
+Input
+
+| Start timestamp | End timestamp |
+| --------------- | ------------- |
+| 0.0             | 0.1           |
+| 1.0             | 1.1           |
+| 2.0             | NaN           |
+| 3.0             | 3.1           |
+| ...             | ...           |
+
+Output
+
+| Start timestamp | Latency |
+| --------------- | ------- |
+| 0.0             | 0.1     |
+| 1.0             | 0.1     |
+| 3.0             | 0.1     |
+| ...             | ...     |
+
+## Response Time
+
+Response time is the amount of time it takes for a system to respond to an input.
+
+As shown above, latency can be calculated if a table can be constructed.
+However, the calculated latency is not suitable for evaluating response time.
+For example, if a sensor is driven at 10 Hz, a latency of up to 100 ms should be considered.
+
+CARET defines latency as the best-case response time and worst-case response time as the value that includes the delay due to cycles, as a more comprehensive approach.
```

Comment:
```suggestion
CARET serves best-case response time and worst-case response time. The former is equal to latency from an input to a corresponding initial output. CARET give the latter one because CARET takes cycle latency into account.
```